tests/unit/services/test_database_prediction_job.py

```diff
@@ -81,6 +83,25 @@ def test_get_featuresets(self, data_interface_mock):
     def test_get_featureset_names(self, data_interface_mock):
         self.assertEqual(type(self.service.get_featureset_names()), list)
 
+    def test_dataclass(self, data_interface_mock):
+        pj_dataclass = PredictionJobDataClass(**prediction_job)
+        self.assertIsInstance(pj_dataclass, PredictionJobDataClass)
+
+    def test_prediction_job(self, data_interface_mock):
+        pj = self.service._create_prediction_job_object(prediction_job)
+        self.assertEqual(pj.__getitem__("id"), prediction_job["id"])
+        pj.__setitem__("id", 50)
+        self.assertEqual(pj.__getitem__("id"), 50)
+
+        with self.assertRaises(AttributeError):
+            pj.__setitem__("non_existing", "can't")
+
+    def test_no_type(self, data_interface_mock):
+        pj_dict = prediction_job.copy()
+        pj_dict.pop("forecast_type")
+        with self.assertRaises(AttributeError):
+            self.service._create_prediction_job_object(pj_dict)
+
```

Comment:
Good job adding the unit tests!
Tip: for the unit tests why try to use this naming convention:
'test_{function_to_test/class}_{aspect_to_test}`

so `test_no_type` would become `test_create_prediction_job_object_missing_attribute`